frontend/azlinux/handle_rpm.go

```diff
@@ -151,20 +180,117 @@ func installBuildDepsPackage(target string, packageName string, w worker, deps m
 	}
 }
 
+// meant to return a run option for mounting all repo state
+func withRepoData(repos []dalec.PackageRepositoryConfig, sOpts dalec.SourceOpts, opts ...llb.ConstraintsOpt) (llb.RunOption, error) {
```

Comment:
Can we split all these repo functions to a separate file?

Reply:
I actually just did this as part of the work I started for Jammy support! But actually, I could move those changes into this PR.